crates/revm/src/db/in_memory_db.rs

```diff
@@ -113,7 +113,7 @@ impl<ExtDB: DatabaseRef> DatabaseCommit for CacheDB<ExtDB> {
                 self.changes.insert(add, Default::default());
                 self.storage.get_mut(&add).map(clear_storage);
             } else {
-                self.insert_change(add, acc.info);
+                self.insert_cache(add, acc.info);
```

Comment:
In current code, it is correct to use `insert_change`,
this part is done as bug fix for cleared storage (selfdestruct), I will change this when i get more free time but for now it is what it is.

Reply:
Foundry ignores `change` field, and that is where new contract is saved, it is a reason why eth tests fail

Reply:
I see, I can look into making foundry use the `change` field.

Reply:
Revert this and rest can be merged